### PR TITLE
[#275] Publish for new Safari

### LINKED ISSUE #275: Try and publish for new Safari
Apparently Apple is finally bringing the standardized WebExtensions API to Safari, which means Safari users should be able to use Tickety-Tick (without us creating a special Safari App Extension implementation). 🙌 

From the Apple macOS Big Sur feature overview:

screenshot

— https://www.apple.com/macos/big-sur/features/#safari

cc @planktonic

### PR DESCRIPTION
**Pull request by pmeinhardt**
To install Tickety-Tick in Safari (requires Xcode to be installed): 🧑‍🚀 

1. Open Safari
2. Enable "Developer > Allow Unsigned Extensions" (is reset when quitting Safari)
3. Build via `yarn build:safari`
4. Open `dist/safari/tickety-tick.app`

The Xcode project references files in the `dist/chrome` directory. :v:

Closes #275.

### CODE REVIEW

File: safari/tickety-tick Extension/Info.plist [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>CFBundleDevelopmentRegion</key>
+	<string>$(DEVELOPMENT_LANGUAGE)</string>
+	<key>CFBundleDisplayName</key>
+	<string>tickety-tick Extension</string>
```
Comment (pmeinhardt) [inline safari/tickety-tick Extension/Info.plist:8]: Maybe update this:

```suggestion
	<string>Tickety-Tick</string>
```

File: script/extract-cert-ou [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+set -o pipefail
+
+if [ $# -lt 1 ]; then
+  echo "usage: $(basename "$0") <certificate-name>"
+  exit 1
+fi
+
+CERT_NAME="$1"
+
+exec security find-certificate -a -c "$CERT_NAME" -p | keytool -printcert | grep 'Owner: ' | ruby -e 'print STDIN.readline.sub("Owner: ", "").split(", ").map { |p| p.split("=") }.to_h["OU"]'
```
Comment (pmeinhardt) [inline script/extract-cert-ou:14]: Maybe clean up that last part with the Ruby script and use `awk` or at least clean Ruby. 😁

File: script/xcodebuild [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+set -o pipefail
+
+cd "$(dirname "$0")/.."
+
+CODE_SIGN_IDENTITY="${1:-Apple Development}"
+DEVELOPMENT_TEAM=$(script/extract-cert-ou "$CODE_SIGN_IDENTITY")
+
+exec xcodebuild \
+  -project safari/tickety-tick.xcodeproj \
+  -scheme tickety-tick build \
```
Comment (pmeinhardt) [inline script/xcodebuild:14]:
```suggestion
  -scheme tickety-tick \
  build \
```

File: safari/tickety-tick/Assets.xcassets/AppIcon.appiconset/Contents.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      "idiom" : "mac",
+      "scale" : "2x",
+      "size" : "512x512"
+    }
+  ],
+  "info" : {
+    "version" : 1,
+    "author" : "xcode"
+  }
+}
```
Comment (pmeinhardt) [inline safari/tickety-tick/Assets.xcassets/AppIcon.appiconset/Contents.json:63]: Maybe we can symlink those PNG files in `safari/tickety-tick/Assets.xcassets/AppIcon.appiconset/` to `dist/chrome` instead?

**Comment by pmeinhardt** [comment]
Darren built the extension from this branch.

We should document some extra steps we figured out together:

> What you’ll need is a signing certificate.
>
> I think the easiest way to create one might be this:
> 
> 1. Open Xcode
> 2. Go to Preferences > Accounts
> 3. Add an account of type “Apple ID”

Thanks a lot @planktonic 💚

**Review by klappradla — APPROVED** [review]
Yay 🙌 

This was build in multiple pairing sessions - good to hit 🌊 

Browser diversity FTW 💚

**Pull request merged**